dfget/locator/locator.go

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright The Dragonfly Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package locator
+
+// SupernodeLocator defines the way how to get available supernodes.
+// Developers can implement their own locator more flexibly , not just get the
+// supernode list from configuration or CLI.
+type SupernodeLocator interface {
+	// Get returns the current selected supernode, it should be idempotent.
+	Get() *Supernode
+
+	// Next chooses the next available supernode for retrying or other
+	// purpose. The current supernode should be set as this result.
+	Next() *Supernode
+
+	// GetGroup returns the group.
+	GetGroup(name string) *SupernodeGroup
```

Comment:
Just out of curiosity, why consider add a supernode group concept here? And the `SupernodeGroup` can only be retrieved for viewing. Is it expected that I can choose a group to use?

Reply:
For example, we can group supernodes by region, business, version and so on. The implementation of SupernodeLocator can select a supernode based on the group. The caller does not need to care about the specific selection logic, but if it is really needed, the supernode list can also be obtained through this interface.